File: backend/app/schemas/acceptance_steps.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.acceptance_steps import AcceptanceStepStatus


class AcceptanceStepCreate(BaseModel):
    step_no: int
    step_name: str
    responsible_id: UUID
    plan_date: date | None = None
    description: str | None = None


class AcceptanceStepUpdate(BaseModel):
    status: AcceptanceStepStatus


class AcceptanceStepRead(BaseModel):
    id: UUID
    phase_id: UUID
    step_no: int
    step_name: str
    responsible_id: UUID
    plan_date: date | None
    description: str | None
    status: AcceptanceStepStatus
    completed_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class AcceptanceStepListRead(BaseModel):
    items: list[AcceptanceStepRead]
    total: int

File: frontend/src/types/acceptanceSteps.ts
export type AcceptanceStepStatus = 'not_started' | 'in_progress' | 'completed';

export interface AcceptanceStepRead {
  completed_at: string | null;
  created_at: string;
  description: string | null;
  id: string;
  phase_id: string;
  plan_date: string | null;
  responsible_id: string;
  status: AcceptanceStepStatus;
  step_name: string;
  step_no: number;
  updated_at: string;
}

export interface AcceptanceStepListRead {
  items: AcceptanceStepRead[];
  total: number;
}

export interface AcceptanceStepCreatePayload {
  description?: string | null;
  planDate?: string | null;
  responsibleId: string;
  stepName: string;
  stepNo: number;
}

export interface AcceptanceStepUpdatePayload {
  status: AcceptanceStepStatus;
}

export const ACCEPTANCE_STEP_STATUS_LABELS: Record<AcceptanceStepStatus, string> = {
  completed: '已完成',
  in_progress: '进行中',
  not_started: '未开始',
};
